src/utils/formatNumbers.test.ts

```diff
@@ -460,3 +460,37 @@ describe('formatReviewSwapCurrencyAmount', () => {
     expect(formatReviewSwapCurrencyAmount(currencyAmount)).toBe('2000000')
   })
 })
+
+describe('formatPercent', () => {
+  beforeEach(() => {
+    mocked(useLocalCurrencyConversionRate).mockReturnValue({ data: 1.0, isLoading: false })
+    mocked(useCurrencyConversionFlagEnabled).mockReturnValue(true)
+  })
+
+  it('should correctly format undefined', () => {
+    const { formatPercent } = renderHook(() => useFormatter()).result.current
+
+    expect(formatPercent(undefined)).toBe('-')
+  })
```

Comment:
you should include all of the special cases here: `null`, `undefined`, `Infinity`, `NaN`.
you can do it most elegantly with `it.each`:

```suggestion
  it.each([null, undefined, Infinity, NaN])('should correctly format %p', ([value]) => {
    const { formatPercent } = renderHook(() => useFormatter()).result.current

    expect(formatPercent(value)).toBe('-')
  })
```

Reply:
oh cool - thats good to know ty